sdk/python/tests/compiler/testdata/conditions_and_loops.yaml

```diff
@@ -33,6 +33,195 @@ metadata:
       [{"key": "artifacts/$PIPELINERUN/print-number/Output.tgz", "name": "print-number-Output",
       "path": "/tmp/outputs/Output/data"}], "produce-numbers": [{"key": "artifacts/$PIPELINERUN/produce-numbers/Output.tgz",
       "name": "produce-numbers-Output", "path": "/tmp/outputs/Output/data"}]}'
+    tekton.dev/resource_templates: "\n---\napiVersion: custom.tekton.dev/v1alpha1\n\
```

Comment:
Hi Tommy, do you think we could convert the resource template string to JSON/Dict here in order to preserve some readability? I don't presume many users will want/need to read this, but for debugging and understanding purposes JSON should be easier

Reply:
@drewbutlerbb4 @fenglixa do you have any preference on JSON vs yaml when parsing strings in the UI? If not, I will convert all the custom task CRs into to list of JSON as @ckadner suggested. JSON also could help us save some character space as well.

Reply:
This annotation has two purpose, 
1. API uses it to deploy the custom task CR before running the pipeline
2. UI able to use this information to display loop dag with more details.

Reply:
@Tomcli, yes, I used the json format handled such situation before, refer to my another PR, FYI:
https://github.com/kubeflow/kfp-tekton/pull/489/files#diff-5aee3b3cd734f4dd317784331d99f1e76d0283631755c31cd67379220e883734R224-R226

Reply:
done, the annotations now store the loop CRs as JSON string